addons/core/addon/decorators/notify.js

```diff
@@ -50,14 +52,19 @@ export function notifySuccess(notification) {
  *
  * @param {string|function} notification
  * @param {object} options
- * @param {object} options.sticky - defaults to true, whether or not to persist
+ * @param {boolean} options.sticky - defaults to true, whether or not to persist
  *                                  the notification until user dismissal
- * @param {object} options.catch - defaults to false, whether or not to catch
+ * @param {boolean} options.catch - defaults to false, whether or not to catch
  *                                 and squelch the error
+ * @param {object} options.log - defaults to undefined, object with unique
+ *                              information about the error to send to our
+ *                              electron logging service
+ * @param {string} options.log.origin - defaults to undefined, the origin that
+ *                                     triggered the error
  */
 export function notifyError(
   notification,
-  options = { catch: false, sticky: true },
+  options = { catch: false, sticky: true, log: undefined },
```

Comment:
Is this `undefined` default necessary?

Reply:
I don't feel like it is but I added it to keep with the pattern. Would you prefer I remove it, or all of them, especially since the defaults are explained in the signature?
Those aren't even true defaults. This is a comment lower in the decorator:

> // Default is `true`.  Since `options` is a new object instance
        // when passed, the values shown in the signature aren't
        // true defaults.  For example, if a value `{ catch: true }` were
        // passed, `sticky` would actually be falsy, even though the intended
        // default is `true`.  Hence this extra step.
        const sticky = options.sticky === undefined ? true : options.sticky;

Reply:
I think the other two are fine since they're booleans and it's fine to default them to actual booleans, but `log` would still be `undefined` either way. I think they're not "true defaults" only in the sense of how JS works. Otherwise I think it's accurate